homeassistant/components/ihc/__init__.py

```diff
@@ -4,68 +4,124 @@
 from ihcsdk.ihccontroller import IHCController
 import voluptuous as vol
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.typing import ConfigType
 
 from .auto_setup import autosetup_ihc_products
 from .const import (
     CONF_AUTOSETUP,
-    CONF_INFO,
     DOMAIN,
     IHC_CONTROLLER,
-    IHC_CONTROLLER_INDEX,
+    IHC_CONTROLLER_ID,
+    IHC_PLATFORMS,
 )
-from .manual_setup import IHC_SCHEMA, get_manual_configuration
+from .manual_setup import MANUAL_SETUP_SCHEMA, manual_setup
+from .migrate import migrate_configuration
 from .service_functions import setup_service_functions
 
 _LOGGER = logging.getLogger(__name__)
 
-
+"""
+CONFIG_SCHEMA is not used by the setup anymore. It is there to make hassfest happy.
+The setup below does not setup the ihc configuration, it is used to migrate the old
+manual setup of ihc devices from configuration.yaml
+"""
 CONFIG_SCHEMA = vol.Schema(
-    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [IHC_SCHEMA]))}, extra=vol.ALLOW_EXTRA
+    {DOMAIN: vol.Schema(vol.All(cv.ensure_list, [MANUAL_SETUP_SCHEMA]))},
+    extra=vol.ALLOW_EXTRA,
 )
 
 
 def setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the IHC integration."""
-    conf = config[DOMAIN]
-    for index, controller_conf in enumerate(conf):
-        if not ihc_setup(hass, config, controller_conf, index):
-            return False
+    if config.get(DOMAIN) is not None:
+        _LOGGER.error(
+            """
+            Setup of the IHC controller in configuration.yaml is no longer
+            supported. See https://www.home-assistant.io/integrations/ihc/
+            """
+        )
```

Comment:
Please create a repair issue where you inform the user that the configuration is migrated, and he can now delete the yaml part.

See also https://github.com/home-assistant/architecture/blob/master/adr/0021-YAML-integration-configuration-deprecation-policy.md